### ISSUE TITLE
Request for additional description of PRECAST model in tutorial/manual

### ISSUE THREAD

**Issue by jac-thom** [feiyoung/PRECAST#20]
Hello,

I think PRECAST has the potential to work for my 10X visium dataset but I have a few questions about how the models are run.

1. In my experience with machine learning models, the loss plots are key to determining model fit. Can you confirm that the number of iterations (which can be specified in the PRECAST model parameters) are equivalent to the # of epochs used to train the model(s)? Can you provide a range of iterations that you think would be helpful if a user is looking to optimize the model? If possible can you describe whether there is any output produced by the PRECAST function log that can help inform the number of iterations to select?
2. I had not heard of energy functions before, but a google search indicates that they are related to loss functions. Can you describe how loss/energy is minimized between epochs (or iterations) and provide a guide for interpreting the 'Diff. Energy' value that is produced by each iteration (when looking at the verbose output of the PRECAST function). 
3. If the log likelihood and 'dloglik' are not related to optimizing loss between epochs/iterations (Q#2), can you please explain how these outputs should be used/ interpreted?
4. I assume that the PRECAST parameter 'Sigma_equal' determines an assumption related to estimating some kind of variance across samples (list elements) input to the model. Can you clarify what this parameter does and perhaps give guidelines for whether a user should set it to T or F for their own data?

Thank you in advance! I know that these are relatively detailed questions but I am hoping to implement PRECAST for a large dataset and want to be more familiar with the under-the-hood mechanics before I commit to doing so.

Best,
Jacqui

**Comment by feiyoung** [comment]
Hi, Jacqui,

Thank you for your attention to our work. Here are some clarifications regarding the aspects you inquired about:

1. Absolutely, the loss plots are crucial for assessing the model's fit. Our objective function is the log-likelihood, which we aim to maximize. For each iteration, the value of the log-likelihood is stored in `PRECASTObj@resList[[k]]$loglik_seq`. The variable `dloglike` represents the relative variation of the log-likelihood, calculated as `(loglike[t] - loglike[t-1]) / abs(loglike[t-1])`. If this variation falls below a certain threshold, the algorithm terminates. Our experience suggests that a maximum of 30 to 50 iterations often works well.

2. The energy function, in the context of the ICM algorithm, can be interpreted as the negative log-likelihood for the parameters of the latent cluster labels. It aids in overcoming challenges posed by spatial dependence and facilitates obtaining posterior estimates of cluster labels.

3. Indeed, the negative log-likelihood can be considered as the loss function in our model.

4. The parameter 'Sigma_equal' determines whether the covariance matrix is equal across different clusters in the Gaussian mixture model used in PRECAST.

We highly recommend reading our supplementary text, which provides detailed information on PRECAST.

Best regards,
Wei Liu

**Comment by jac-thom** [comment]
Thank you very much for this info, I will take a closer look at the supplemental text for future questions!
